### PR TITLE
fixed more tests and made mocks follow new vmcommon interface

### PR DESCRIPTION
**Pull request by ccorcoveanu**
Integrated new vmcommon + removed references to IELE for now.
Some tests are still failing but unrelated to to the new vmcommon.

New PR will follow integrating arwen promise implementation

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -56,7 +61,10 @@ github.com/coreos/go-etcd v2.0.0+incompatible/go.mod h1:Jez6KQU2B/sWsbdaef3ED8Nz
 github.com/coreos/go-semver v0.2.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
 github.com/coreos/go-semver v0.3.0 h1:wkHLiw0WNATZnSG7epLsujiMCgPAc9xhjJ4tgnAxmfM=
 github.com/coreos/go-semver v0.3.0/go.mod h1:nnelYz7RCh+5ahJtPPxZlU+153eP4D4r3EedlOD2RNk=
+github.com/cpuguy83/go-md2man v1.0.10 h1:BSKMNlYxDvnunlTymqtgONjNnaRV1sTpcovwwjF22jk=
 github.com/cpuguy83/go-md2man v1.0.10/go.mod h1:SmD6nW6nTyfqj6ABTjUi3V3JVMnlJmwcJI5acqYI6dE=
+github.com/cpuguy83/go-md2man/v2 v2.0.0-20190314233015-f79a8a8ca69d h1:U+s90UTSYgptZMwQh2aRr3LuazLJIa+Pg3Kc1ylSYVY=
```
Comment (sasurobert) [inline go.sum:66]: go mod tidy

File: integrationTests/vm/iele/vmAgar/vmAgar_test.go [hunk 2]
```diff
@@ -284,7 +284,8 @@ func TestAgarioContractJoinGameReward(t *testing.T) {
 	_, err = testContext.Accounts.Commit()
```
Comment (sasurobert) [inline integrationTests/vm/iele/vmAgar/vmAgar_test.go:284]: delete folder IELE

File: core/converters.go [hunk 3]
```diff
@@ -162,3 +162,11 @@ func ConvertToEvenHexBigInt(value *big.Int) string {
 
 	return str
 }
+
+// BooleanToInt takes a bool argument and returns 0 for false, 1 for true
+func BooleanToInt(b bool) int {
```
Comment (iulianpascalau) [inline core/converters.go:167]: Not tested

File: process/smartContract/process_test.go [hunk 4]
```diff
@@ -1298,7 +1298,53 @@ func TestScProcessor_RefundGasToSender(t *testing.T) {
 	)
 	require.Nil(t, err)
 
+	finalValue := big.NewInt(0).Mul(refundGas, big.NewInt(0).SetUint64(sc.economicsFee.MinGasPrice()))
 	require.Equal(t, currBalance, acntSrc.(state.UserAccountHandler).GetBalance().Uint64())
+	require.Equal(t, scr.Value.Cmp(finalValue), 0)
+}
+
+func TestScProcessor_DoNotRefundGasToSenderForAsyncCall(t *testing.T) {
```
Comment (iulianpascalau) [inline process/smartContract/process_test.go:1306]: :+1:

**Pull request merged**